cmd/influx/importer/v8.go

```diff
@@ -0,0 +1,222 @@
+package importer
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/influxdb/influxdb/client"
+)
+
+const batchSize = 5000
+
+// V8Config is the config used to initialize a V8 importer
+type V8Config struct {
+	username, password string
+	url                url.URL
+	precision          string
+	writeConsistency   string
+	file, version      string
+	compressed         bool
+}
+
+// NewV8Config returns an initialized *V8Config
+func NewV8Config(username, password, precision, writeConsistency, file, version string, u url.URL, compressed bool) *V8Config {
+	return &V8Config{
+		username:         username,
+		password:         password,
+		precision:        precision,
+		writeConsistency: writeConsistency,
+		file:             file,
+		version:          version,
+		url:              u,
+		compressed:       compressed,
+	}
+}
+
+// V8 is the importer used for importing 0.8 data
+type V8 struct {
+	client                                     *client.Client
+	database                                   string
+	retentionPolicy                            string
+	config                                     *V8Config
+	wg                                         sync.WaitGroup
+	line, command                              chan string
+	done                                       chan struct{}
+	batch                                      []string
+	totalInserts, failedInserts, totalCommands int
+}
+
+// NewV8 will return an intialized V8 struct
+func NewV8(config *V8Config) *V8 {
+	return &V8{
+		config:  config,
+		done:    make(chan struct{}),
+		line:    make(chan string),
+		command: make(chan string),
+		batch:   make([]string, 0, batchSize),
+	}
+}
+
+// Import processes the specified file in the V8Config and writes the data to the databases in chukes specified by batchSize
+func (v8 *V8) Import() error {
+	// Create a client and try to connect
+	config := client.NewConfig(v8.config.url, v8.config.username, v8.config.password, v8.config.version, client.DEFAULT_TIMEOUT)
+	cl, err := client.NewClient(config)
+	if err != nil {
+		return fmt.Errorf("could not create client %s", err)
+	}
+	v8.client = cl
+	if _, _, e := v8.client.Ping(); e != nil {
+		return fmt.Errorf("failed to connect to %s\n", v8.client.Addr())
+	}
+
+	// Validate args
+	if v8.config.file == "" {
+		return fmt.Errorf("file argument required")
+	}
+
+	defer func() {
+		v8.wg.Wait()
```

Comment:
makes sense.  I'll add the wait calls above this as there are always only 2.  It may refactor out though if I stop using channels and just use straight method calls.  I don't think I gained any performance with the go routines as the read is io bound and the accumulator is memory bound.